src/rdb.c

```diff
@@ -2178,15 +2193,26 @@ robj *rdbLoadObject(int rdbtype, rio *rdb, sds key, int dbid, int *error) {
                 return NULL;
             }
 
+            /* Also load the entry expiry */
+            long long itemexpiry = EXPIRY_NONE;
+            if (rdbtype == RDB_TYPE_HASH_2) {
+                itemexpiry = rdbLoadMillisecondTime(rdb, RDB_VERSION);
+                if (itemexpiry == LLONG_MAX && rioGetReadError(rdb)) return NULL;
```

Comment:
```suggestion
                if (rioGetReadError(rdb)) return NULL;
```
Couldn't someone legit set LLONG_MAX? The comment says you just use the rioGetReadError, and all the other functions just use this.

Reply:
Great catch! 
I really have no idea how this logic came in??? will fix that.